crates/services/executor/src/executor.rs

```diff
@@ -648,16 +680,72 @@ where
                             .events
                             .push(ExecutorEvent::MessageImported(message));
                     }
-                    Event::Transaction(_) => {
-                        // TODO: implement handling of forced transactions in a later PR
+                    Event::Transaction(relayed_tx) => {
+                        let id = relayed_tx.id();
+                        // perform basic checks
+                        let checked_tx_res = Self::validate_forced_tx(
+                            relayed_tx,
+                            header,
+                            &self.consensus_params,
+                        );
+                        // handle the result
+                        match checked_tx_res {
+                            Ok(checked_tx) => {
+                                checked_forced_txs.push(checked_tx);
+                            }
+                            Err(err) => {
+                                execution_data.events.push(
+                                    ExecutorEvent::ForcedTransactionFailed {
+                                        id,
+                                        block_height,
+                                        block_time: header.consensus.time,
+                                        failure: err.to_string(),
+                                    },
+                                );
+                            }
+                        }
                     }
                 }
             }
         }
 
         execution_data.event_inbox_root = root_calculator.root().into();
 
-        Ok(())
+        Ok(checked_forced_txs)
+    }
+
+    /// Parse forced transaction payloads and perform basic checks
+    fn validate_forced_tx(
+        relayed_tx: RelayedTransaction,
+        header: &PartialBlockHeader,
+        consensus_params: &ConsensusParameters,
+    ) -> Result<MaybeCheckedTransaction, ForcedTransactionFailure> {
+        let parsed_tx = Transaction::from_bytes(relayed_tx.serialized_transaction())
+            .map_err(|_| ForcedTransactionFailure::CodecError)?;
+
+        let check_tx_res = parsed_tx
+            .into_checked(header.consensus.height, consensus_params)
+            .map_err(ForcedTransactionFailure::CheckError)?;
+
+        let checked_tx = check_tx_res
```

Comment:
Before we run any predicates, we need to check that the calculated `max_gas <= relayed_max_gas`.